### PR TITLE
refactor iceAgent and iceAgentStateMachine to make the state behavior…

### PR DESCRIPTION
**Pull request by chehefen**
… more clear

*Issue #, if available:*

*Description of changes:*
refactor iceAgent and iceAgentStateMachine to make the state behavior more clear

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Ice/IceAgent.c [hunk 1]
```diff
@@ -859,7 +859,7 @@ STATUS iceAgentStateNewTimerCallback(UINT32 timerId, UINT64 currentTime, UINT64
     MUTEX_LOCK(pIceAgent->lock);
     locked = TRUE;
 
-    // prime the state machine
+    // drive the state machine
```
Comment (MushMal) [inline src/source/Ice/IceAgent.c]: Comment for LN 869 as it won't allow me to comment there. I think the best would be to allow the state machine transaction itself to lead to the failed state. So, for example, stepIceAgentStateMachine if failed itself, should result in overall failure. It should still return SUCCESS even if logically moving to an FATAL error state.

Currently, the iceAgentFatalError function simply assigns an error to the status. Perhaps this is the correct action itself.

Comment (chehefen) [reply src/source/Ice/IceAgent.c]: I am gonna address that in a separate PR cuz this one is getting quite lengthy.

File: src/source/Ice/IceAgent.c [hunk 2]
```diff
@@ -877,31 +877,16 @@ STATUS iceAgentStateNewTimerCallback(UINT32 timerId, UINT64 currentTime, UINT64
...
-    PIceAgent pIceAgent = (PIceAgent) customData;
-    BOOL locked = FALSE;
     PDoubleListNode pCurNode = NULL;
     UINT64 data;
     PIceCandidate pCandidate = NULL;
     PIceServer pIceServer = NULL;
 
     CHK(pIceAgent != NULL, STATUS_NULL_ARG);
 
-    MUTEX_LOCK(pIceAgent->lock);
```
[the remark this answers is not in the document]
Comment (chehefen) [reply src/source/Ice/IceAgent.c]: done

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189?src=pr&el=h1) Report
> Merging [#189](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/commit/a805037cac46791915bde09f2365c0559f6ae917&el=desc) will **increase** coverage by `0.10%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #189      +/-   ##
==========================================
+ Coverage   86.90%   87.00%   +0.10%     
==========================================
  Files          33       33              
  Lines        7644     7649       +5     
==========================================
+ Hits         6643     6655      +12     
+ Misses       1001      994       -7     
```


| [Impacted Files](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/source/Ice/IceAgent.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnQuYw==) | `93.81% <0.00%> (+0.56%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189?src=pr&el=footer). Last update [2dce050...9037d8b](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/189?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**